clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/Replacer.java

```diff
@@ -83,7 +63,8 @@ private String nameFromReference(String s) {
 
   @Nonnull
   ImmutableCollection<Artifact> getArtifacts(DocumentContext document) {
-    return mapper.convertValue(findAll(document), new TypeReference<List<String>>() {}).stream()
+    return Streams.stream(document.<ArrayNode>read(findPath).elements())
```

Comment:
> This is much clearer (and will perform better) than using an ObjectMapper for this.

Agree it is clearer, but curious how you knew it would perform better?

Reply:
Good point; I didn't actually profile this particular call before/after so I can't say for sure.  I probably should have said "likely will perform better".

The reason I think it will perform better is that in general serialization/deserialization/conversion with `ObjectMapper` is pretty slow because it uses reflection to iterate over all properties (or builders/setters, depending on how it's configured) of the object and the target type to do the conversion.

Here since we're replacing that with looping over the found elements and calling a single `getText` function on each I'd expect it to be much faster.   In my profiling this conversion wasn't the bottleneck (compiling the JsonPath was), so it's probably not super important.  But anyway that's why I'd expect this to be faster.

Reply:
💯 thank you for explaining! (wasn't so much worried about actual performance improvements from this, just want to develop a better sense for these things like you have)